scorex-basics/src/main/scala/scorex/network/HistorySynchronizer.scala

```diff
@@ -172,10 +185,11 @@ class HistorySynchronizer(application: Application) extends ViewSynchronizer wit
     syncing
   }
 
-  private def gotoGettingExtension(betterScore: BigInt, witnesses: Seq[ConnectedPeer]): Unit = {
+  private def gotoGettingExtension(witnesses: Seq[ConnectedPeer]): Unit = {
     log.debug("Transition to gettingExtension")
     blockGenerator ! StopGeneration
-    context become gettingExtension(betterScore, witnesses)
+    context become gettingExtension(witnesses)
+    gettingExtension(witnesses)
```

Comment:
Well, I just tried to find a cause of the problem I have fixed, and I have decided to add this line as for other "goto"s, and since all tests (including test net integration) are passed. 

But, TBH, for me this is the most confusion part of Scorex - why to call the a receive function while processing a message?  We have become() call anyway changing behaviour in this way.